### PR TITLE
add support for storing data in kube secrets

### PR DESCRIPTION
**Pull request by sidpremkumar**
## Description of the change

Adds secret support to the kube agent

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] Refactor (code change with no functionality change)
- [ ] Enhancement (minor change below the level of a feature)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Remove feature (removes a feature we don't support anymore)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related JIRA tickets

Relates to JIRA: CWC-995

## Relevant release note information

Release Notes:

## Potential Security Impacts

Does this PR have any security impact?
Does it fix a security issue?
Does it add attack surface?
Why do we think this change is safe?

### Development

- [ ] Does the code changed/added as part of this pull request have test coverage?
- [ ] Do all tests related to the changed code pass in development?
- [ ] Have you ensured that at least one other person has tested your code?
- [ ] Have you tested the code?
- [ ] Have you attached any pictures (if relevant)?

### CODE REVIEW

File: bctl-go-daemon/Server/Websockets/controlWebsocket/controlWebsocket.go [hunk 1]
```diff
@@ -36,6 +36,20 @@ func NewControlWebsocketClient(serviceURL string, activationToken string, orgId
 
 	ret.WebsocketClient = commonWebsocketClient.NewCommonWebsocketClient(serviceURL, hubEndpoint, params, headers)
 
+	// Determin if this is a fresh agent
+	if (ret.NewAgent() == true){
+		// Update our persistant secret
+		secretInfo := controlWebsocketTypes.SecretConfig{}
+		secretInfo.Test = "testwow"
```
Comment (tflpd) [inline bctl-go-daemon/Server/Websockets/controlWebsocket/controlWebsocket.go:43]: cool beans**

File: bctl-go-daemon/Server/Websockets/controlWebsocket/controlWebsocketTypes/controlWebsocketTypes.go [hunk 2]
```diff
@@ -75,3 +88,104 @@ func (client *ControlWebsocket) SendAliveCheckToBastionFromClusterMessage(aliveC
...
+
+func (client *ControlWebsocket) GetSecretClient() coreV1Types.SecretInterface {
+	// Create our api object
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		panic(err.Error())
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err.Error())
```
Comment (tflpd) [inline bctl-go-daemon/Server/Websockets/controlWebsocket/controlWebsocketTypes/controlWebsocketTypes.go:162]: Maybe I m wrong but I think `log.Fatal(err)` panics while providing a more detailed error msg - maybe I am wrong though

**Review by tflpd — APPROVED** [review]

File: bctl-go-daemon/Server/Websockets/controlWebsocket/controlWebsocketTypes/controlWebsocketTypes.go (hunk 2 above)
Comment (sidpremkumar) [reply bctl-go-daemon/Server/Websockets/controlWebsocket/controlWebsocketTypes/controlWebsocketTypes.go:162]: @lgmugnier or @ymarcus93 I will defer this q to one of you

Comment (ymarcus93) [reply bctl-go-daemon/Server/Websockets/controlWebsocket/controlWebsocketTypes/controlWebsocketTypes.go:162]: `log.Fatal` immediately exits the program. You can technically `recover` from a `panic` if you so desire. `panic` gives stack trace + message; `log.Fatal` just prints the message (in this case an error message). `panic` will still run the `defer` functions as well.

Comment (sidpremkumar) [reply bctl-go-daemon/Server/Websockets/controlWebsocket/controlWebsocketTypes/controlWebsocketTypes.go:162]: So in this case it sounds like `panic` is what we want? We would want it to exit the program and give us a message and run any pending `defer`s. Does that sound correct?

File: bctl-go-daemon/Server/Websockets/controlWebsocket/controlWebsocket.go (hunk 1 above)
Comment (sidpremkumar) [reply bctl-go-daemon/Server/Websockets/controlWebsocket/controlWebsocket.go]: https://github.com/bastionzero/zli/pull/134/commits/3e1505dee25da0c7fadbd6f4fda503297586ef65 :)

File: bctl-go-daemon/Server/Websockets/controlWebsocket/controlWebsocket.go [hunk 3]
```diff
@@ -40,7 +40,7 @@ func NewControlWebsocketClient(serviceURL string, activationToken string, orgId
 	if (ret.NewAgent() == true){
 		// Update our persistant secret
 		secretInfo := controlWebsocketTypes.SecretConfig{}
-		secretInfo.Test = "testwow"
+		secretInfo.Test = "coolbeans"
```
Comment (ymarcus93) [inline bctl-go-daemon/Server/Websockets/controlWebsocket/controlWebsocket.go:43]: lol

File: bctl-go-daemon/Server/Websockets/controlWebsocket/controlWebsocketTypes/controlWebsocketTypes.go (hunk 2 above)
Comment (ymarcus93) [reply bctl-go-daemon/Server/Websockets/controlWebsocket/controlWebsocketTypes/controlWebsocketTypes.go:162]: In general, it's better to not `panic`. Although it's slightly verbose (welcome to Go's way of error handling), returning an `error` as the second value in this function would be common practice. That way the `error` can bubble up all the way to the top of the chain of function calls (in this case all the way up to the daemon server) and then at the top you can decide what todo. The nice thing of having it bubble up is there might be some cleanup each method in the chain of functions has todo.

For example, at the top you might want to log the error message and then do a graceful shutdown of the server via [`Server.Shutdown`](https://pkg.go.dev/net/http?utm_source=godoc#Server.Shutdown). Or for example, you might want to cancel some background goroutine that is doing some processing.

Basically, a `panic` crashes the app fatally and doesn't let graceful shutdown/cleanup to happen--unless of course you `recover` in `main` but I wouldn't do that...

**Review by lgmugnier — APPROVED** [review]
ty!

File: bctl-go-daemon/Server/Websockets/controlWebsocket/controlWebsocketTypes/controlWebsocketTypes.go (hunk 2 above)
Comment (sidpremkumar) [reply bctl-go-daemon/Server/Websockets/controlWebsocket/controlWebsocketTypes/controlWebsocketTypes.go:162]: I've tasked @lgmugnier with coming up with our logging strategy. I will leave this the way it is now, but we have to come back and update all these references across the entire application. So I would rather do that at once and standard than across different PRs differently.
